doc/source/troubleshooting.rst

```diff
@@ -0,0 +1,104 @@
+
+Users should ultimately check ``covalent logs`` for more information and submit a new issue on Github `discussion <https://github.com/AgnostiqHQ/covalent/issues>`_ with any relevant log information associated with the issue.
+
+------------------------------------------
+Covalent CLI commands throws error/warning
+------------------------------------------
+
+1. The covalent config file can periodically get corrupted due to multiple processes attempting to modify the config file simultaneously. This can sometimes be fixed by manually editing the config file. However, if this does not work, the user can delete the config file and restart the Covalent server.
+
+.. warning::
+
+    The user also has the option of running ``covalent purge`` which will delete the config file. A new config file will be created when the user runs ``covalent start`` again. This option must be used with caution.
+
+2. If DB migration error is thrown, that implies that the database schema is not up to date with the latest version of Covalent. This can be fixed by running ``covalent db migrate``.
+
+
+-----------------------------------------
+Getting Result fails for long workflows
+-----------------------------------------
+
+1. For long-running workflows if a user runs ``get_result`` synchronously with ``wait=True`` and observes a ``RecursionError: maximum recursion depth exceeded`` error this means that the result may still be pending or complete but covalent failed during the polling process. Users should still be able to re-run the command to continue waiting for a workflow result.
+
+
+
+----------------------------------
+Executor issues after installation
+----------------------------------
+
+1. Users can get ``executor not found`` or ``Covalent config file missing default values`` for the executor if Covalent was not restarted after the executor was installed. This can be fixed by restarting the Covalent server.
+
+
+-------------------------------------------------------
+Lattice not found error when using Self-hosted Covalent
+-------------------------------------------------------
+
+1. Errors related to the lattice not being found arise from the user trying to retrieve / access data corresponding to a dispatch id that does not exist in the database. This can happen when the self-hosted and local Covalent servers get mixed up. This can be avoided by explicitly specifying the dispatcher address when dispatching and retrieving results.
+
+.. note::
+
+    In general, users should modify the ``dispatcher address`` in the ``ct.dispatch()/ct.get_result()`` functions rather than modifying the config file directly.
```

Comment:
Is `dispatcher address` meant to be `dispatcher_addr`? or to be used as "dispatcher address"? If it's in a code block then it should be a variable name no?

>In general, users should modify the ``dispatcher address`` in the ``ct.dispatch()/ct.get_result()`` functions rather than modifying the config file directly.

I would say maybe something like "... users should pass `dispatcher_addr` (or whatever the arg name is) as argument to `ct.dispatch()/ct.get_result()` functions rather than using `ct.set_config` if they'd like to only temporarily change the address ..." would be better.